tf-psa-crypto/drivers/builtin/src/ccm.c

```diff
@@ -172,6 +172,7 @@ static int ccm_calculate_first_block_if_ready(mbedtls_ccm_context *ctx)
     if (ctx->tag_len == 0) {
         if (ctx->mode == MBEDTLS_CCM_STAR_ENCRYPT || ctx->mode == MBEDTLS_CCM_STAR_DECRYPT) {
             ctx->plaintext_len = 0;
+            return ret;
```

Comment:
We should update the comment. We're not just ignoring the plaintext length: we're skipping the tag calculation.